src/coreComponents/fieldSpecification/FieldSpecificationManager.cpp

```diff
@@ -194,20 +194,62 @@ void FieldSpecificationManager::validateBoundaryConditions( MeshLevel & mesh ) c
 
     if( isFieldNameFound == 0 )
     {
-      char const fieldNameNotFoundMessage[] =
-        "\n{}: there is no {} named `{}` under the {} `{}`.\n";
-      string const errorMsg =
-        GEOS_FMT( fieldNameNotFoundMessage,
+      std::ostringstream fieldNameNotFoundMessage;
+      std::string fieldNamePath =
+        GEOS_FMT( "\n{}: there is no {} named `{}` under the {} `{}`.\n",
                   fs.getWrapperDataContext( FieldSpecificationBase::viewKeyStruct::fieldNameString() ),
                   FieldSpecificationBase::viewKeyStruct::fieldNameString(),
-                  fs.getFieldName(), FieldSpecificationBase::viewKeyStruct::objectPathString(), fs.getObjectPath() );
+                  fs.getFieldName(), FieldSpecificationBase::viewKeyStruct::objectPathString(), fs.getObjectPath());
+
+      fieldNameNotFoundMessage << fieldNamePath;
       if( areAllSetsEmpty )
       {
-        GEOS_LOG_RANK_0( errorMsg );
+        GEOS_LOG_RANK_0( fieldNameNotFoundMessage.str() );
       }
       else
       {
-        GEOS_THROW( errorMsg, InputError );
+        fieldNameNotFoundMessage << GEOS_FMT( "Available fields in {} are:\n", fs.getObjectPath());
+        std::set< string > fieldNameAvail;
+        this->forSubGroups< FieldSpecificationBase >( [&] ( FieldSpecificationBase const & fs2 )
+        {
+          fs2.apply< dataRepository::Group >( mesh,
+                                              [&]( FieldSpecificationBase const &,
+                                                   string const & setName,
+                                                   SortedArrayView< localIndex const > const & targetSet,
+                                                   Group & targetGroup,
+                                                   string const fieldName )
+          {
+            GEOS_UNUSED_VAR( setName );
+            GEOS_UNUSED_VAR( isTargetSetCreated );
+            GEOS_UNUSED_VAR( targetSet );
+
```

Comment:
You can shorter this by using unnamed parameters, which mark these parameters as _unused_.
```suggestion
          fs2.apply< dataRepository::Group >( mesh,
                                              [&]( FieldSpecificationBase const &,
                                                   string const &,
                                                   SortedArrayView< localIndex const > const &,
                                                   Group & targetGroup,
                                                   string const fieldName )
          {
```

Also, `isTargetSetCreated` seems to be used actually, isn't it?

Reply:
yep removed